api/app/controllers/spree/api/v2/shipments_controller.rb

```diff
@@ -0,0 +1,155 @@
+module Spree
+  module Api
+    module V2
+      class ShipmentsController < Spree::Api::BaseController
+
+        before_action :find_and_update_shipment, only: [:ship, :ready, :add, :remove]
+        before_action :load_transfer_params, only: [:transfer_to_location, :transfer_to_shipment]
+
+        def mine
+          if current_api_user.persisted?
+            @shipments = Spree::Shipment
+              .reverse_chronological
+              .joins(:order)
+              .where(spree_orders: {user_id: current_api_user.id})
+              .includes(mine_includes)
+              .ransack(params[:q]).result.page(params[:page]).per(params[:per_page])
+
+            render json: @shipments, meta: pagination(@shipments)
+          else
+            render json: { error: I18n.t(:unauthorized, scope: "spree.api") },
+              status: 401 and return
+          end
+        end
+
+        def create
+          @order = Spree::Order.find_by!(number: params.fetch(:shipment).fetch(:order_id))
+          authorize! :read, @order
+          authorize! :create, Shipment
+          quantity = params[:quantity].to_i
+          @shipment = @order.shipments.create(stock_location_id: params.fetch(:stock_location_id))
+          @order.contents.add(variant, quantity, {shipment: @shipment})
+
+          @shipment.save!
+
+          render json: @shipment
+        end
+
+        def update
+          @shipment = Spree::Shipment.accessible_by(current_ability, :update).readonly(false).friendly.find(params[:id])
+          @shipment.update_attributes_and_order(shipment_params)
+
+          render json: @shipment
+        end
+
+        def ready
+          unless @shipment.ready?
+            if @shipment.can_ready?
+              @shipment.ready!
+            else
+              render json: { error: I18n.t(:cannot_ready, scope: "spree.api.shipment") },
+                status: 422 and return
+            end
+          end
+          render json: @shipment
+        end
+
+        def ship
+          unless @shipment.shipped?
+            @shipment.ship!
+          end
+          render json: @shipment
+        end
+
+        def add
+          quantity = params[:quantity].to_i
+
+          @shipment.order.contents.add(variant, quantity, {shipment: @shipment})
+
+          render json: @shipment
+        end
+
+        def remove
+          quantity = params[:quantity].to_i
+
+          @shipment.order.contents.remove(variant, quantity, {shipment: @shipment})
```

Comment:
Line is too long. [83/80]<br>Redundant curly braces around a hash parameter.<br>Space inside { missing.<br>Space inside } missing.